### DISCUSSION TITLE
How to extend Administration interface with sulu 2.5 ?

### DISCUSSION THREAD

**Discussion by GandonNicolas** [sulu/sulu#6792] [Q&A] [answered]
Hello,

I can't find how with sulu 2.5 to create a JS bundle to extend the administration interface. 
In my old sulu application (1.6) I was using `WebsocketBundle` but I saw that since sulu `2.0.0-alpha4` it has been removed. 

Can someone help me understand how to do this now?

Thanks.

**Comment by alexander-schranz** [comment, marked as answer]
Not sure what you did with the `WebsocketBundle` we are not longer using Websockets or provide a Websocket Server. We rely on basic requester via Javascript fetch method to communicate with the APIs.

To get into Sulu and how to work with Sulu 2.x I recommend doing the `Sulu-Workshop` (assignment-by-assignment yourself) and read the `Extend Admin UI` Docs.

 - [Sulu Workshop](https://github.com/sulu/sulu-workshop)
 - [Sulu Extend Admin](https://docs.sulu.io/en/2.5/book/extend-admin.html)
 
In 2.x in most cases you don't longer need custom JS like it was in 1.6 most basic things like, `List`, `Forms`, ... can easly be added without a single line of JavaScript, but you will get into that doing the Sulu Workshop.

Beside the above we also have some `Tutorials` and Examples on the Sulu Demo:

 - [Sulu Demo Example Pull Requests](https://github.com/sulu/sulu-demo/pulls?q=is%3Aopen+is%3Apr+label%3AExample)

**Reply by GandonNicolas** [reply]
Thanks for the quick response!

I am in the process of upgrading my 1.6 application to 2.5.

In my old application I was only using custom JS for all the admin interface management, forms / lists ....

I would like to not lose this work and all the logic written in JS and be able to reuse it in sulu 2.5.

**Reply by alexander-schranz** [reply]
You can not reuse it. The whole frontend stack is different as we migrated from Backbone JS to React.

For upgrading a project have a look at https://docs.sulu.io/en/2.4/upgrades/upgrade-1.6-2.0.html
But as written above in most cases you don't need any JS anymore as you can do most of Admin with List and Form which do require 0 JS in Sulu 2.x. I really recommend you doing first the Sulu 2 workshop and read then the extend admin docs. To get a feeling how to create custom views and things in Sulu 2 and then do the Upgrade of yourr project.

For all other cases where you had create maybe a special field type you need to reimplement that new field type in `React` JS, there is nearly no way to reuse the `Backbone` components as this are 2 complete different things and most code of 1.6 Bundles is even not longer needed. You see it in the additional `Core` Bundles (Article, Form, Community, Redirect, ...) require now 0 JavaScript in 2.x where they all had a lot of JS in 1.6.

**Reply by GandonNicolas** [reply]
Okay, thank you for all this information 👍
